<?xml version="1.0"?>
<launch>
  <arg name="LOG_LEVEL" default="Info"/>
  <arg name="debug" default="false"/>

  <arg unless="$(arg debug)" name="LAUNCH_PREFIX" value=""/>
  <arg     if="$(arg debug)" name="LAUNCH_PREFIX" value="gdb -ex run --args "/>

  <!-- Orocos target (gnulinux, xenomai, etc) -->
<!--  <arg name="OROCOS_TARGET" default="$(optenv OROCOS_TARGET)"/>-->
  <!-- Path to look for dynamically-loaded plugins and components (something like /lib/orocos) -->
<!--  <arg name="RTT_COMPONENT_PATH" default="$(env RTT_COMPONENT_PATH)"/>-->

  <node
    launch-prefix="$(arg LAUNCH_PREFIX)"
    name="velma_core_cs"
    pkg="subsystem_deployer" type="deployer"
    args="--DeployerName velma_core_cs -l $(arg LOG_LEVEL) -x $(find velma_core_cs)/config/velma_core_cs_transition.xml -m velma_core_cs --" output="screen">
<!--    <env name="OROCOS_TARGET" value="$(arg OROCOS_TARGET)"/>
    <env name="RTT_COMPONENT_PATH" value="$(arg RTT_COMPONENT_PATH)"/>-->
<!--    <env name="LD_PRELOAD" value="libmalloc_hook.so" />-->
  </node>

  <group ns="velma_core_cs">
    <rosparam>
      RightForceTransformation:
        is_right_turn_frame: true
        tool_weight: 12.391
        gravity_arm_in_wrist:
          x: 0.0
          y: 0.0
          z: 0.08
        sensor_frame:
          orientation:
            w: 1.0
            x: 0.0
            y: 0.0
            z: 0.0
          position:
            x: 0.0
            y: 0.0
            z: 0.0
      LeftForceTransformation:
        is_right_turn_frame: true
        tool_weight: 12.391
        gravity_arm_in_wrist:
          x: 0.0
          y: 0.0
          z: 0.08
        sensor_frame:
          orientation:
            w: 1.0
            x: 0.0
            y: 0.0
            z: 0.0
          position:
            x: 0.0
            y: 0.0
            z: 0.0
      HeadIK:
        v_rot:  0.800
        v_lean: 0.375
        v_head: 0.392
        h_cam:  0.0
        v_cam:  0.225
      VL_hp:
        max_vel: 50.0
      VL_ht:
        max_vel: 50.0
      JntImp:
        initial_stiffness: [1200.01, 200.01, 200.01, 200.01, 200.01, 200.01, 200.01, 200.01, 200.01, 200.01, 200.01, 200.01, 200.01, 200.0, 200.01]
      JntLimit:
        upper_limit: [1.56, 2.96, 2.09, 2.96, 2.09, 2.96, 2.09, 2.96, 2.96, 2.09, 2.96, 2.09, 2.96, 2.09, 2.96]
        lower_limit: [-1.56, -2.96, -2.09, -2.96, -2.09, -2.96, -2.09, -2.96, -2.96, -2.09, -2.96, -2.09, -2.96, -2.09, -2.96]
        limit_range: [0.26, 0.26, 0.26, 0.26, 0.26, 0.26, 0.26, 0.26, 0.26, 0.26, 0.26, 0.26, 0.26, 0.26, 0.26]
        max_trq: [10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 6.0, 6.0, 10.0, 10.0, 10.0, 10.0, 10.0, 6.0, 6.0]
      INT_tool_r:
        activate_pose_init: true
        init_setpoint:
          orientation:
            w: 1.0
            x: 0.0
            y: 0.0
            z: 0.0
          position:
            x: 0.0
            y: 0.0
            z: 0.0
        check_tolerances: false
      INT_tool_l:
        activate_pose_init: true
        init_setpoint:
          orientation:
            w: 1.0
            x: 0.0
            y: 0.0
            z: 0.0
          position:
            x: 0.0
            y: 0.0
            z: 0.0
        check_tolerances: false

      wcc_r:
        d0: 0.25
        joint0_idx: 6
        joint1_idx: 7
        constraint_polygon: [-0.397855401039 , -2.90307354927 ,1.79 , -2.91307354927 ,1.78 , -1.43123674393 ,0.77621114254 , -1.39720571041 ,0.36 , -1.00585031509 ,0.35 , 0.414940297604 ,0.8 , 0.942419290543 ,1.8 , 1.01898884773 ,1.81 , 2.88 ,-0.4 , 2.89 ,-0.81 , 2.27813267708 ,-1.82 , 2.29514837265 ,-1.83 , -1.66945314407 ,-0.84 , -1.73751521111 ,-0.423378348351 , -2.09483933449]
      wcc_l:
        d0: 0.25
        joint0_idx: 13
        joint1_idx: 14
        constraint_polygon: [0.397855401039 , 2.90307354927 ,-1.79 , 2.91307354927 ,-1.78 , 1.43123674393 ,-0.77621114254 , 1.39720571041 ,-0.36 , 1.00585031509 ,-0.35 , -0.414940297604 ,-0.8 , -0.942419290543 ,-1.8 , -1.01898884773 ,-1.81 , -2.88 ,0.4 , -2.89 ,0.81 , -2.27813267708 ,1.82 , -2.29514837265 ,1.83 , 1.66945314407 ,0.84 , 1.73751521111 ,0.423378348351 , 2.09483933449]
      ColDet:
        activation_dist: 0.04
        joint_names: [torso_0_joint, right_arm_0_joint, right_arm_1_joint, right_arm_2_joint, right_arm_3_joint, right_arm_4_joint, right_arm_5_joint, right_arm_6_joint, left_arm_0_joint, left_arm_1_joint, left_arm_2_joint, left_arm_3_joint, left_arm_4_joint, left_arm_5_joint, left_arm_6_joint, head_pan_joint, head_tilt_joint, right_HandFingerOneKnuckleOneJoint, right_HandFingerOneKnuckleTwoJoint, right_HandFingerOneKnuckleThreeJoint, right_HandFingerTwoKnuckleOneJoint, right_HandFingerTwoKnuckleTwoJoint, right_HandFingerTwoKnuckleThreeJoint, right_HandFingerThreeKnuckleTwoJoint, right_HandFingerThreeKnuckleThreeJoint, left_HandFingerOneKnuckleOneJoint, left_HandFingerOneKnuckleTwoJoint, left_HandFingerOneKnuckleThreeJoint, left_HandFingerTwoKnuckleOneJoint, left_HandFingerTwoKnuckleTwoJoint, left_HandFingerTwoKnuckleThreeJoint, left_HandFingerThreeKnuckleTwoJoint, left_HandFingerThreeKnuckleThreeJoint ]
    </rosparam>

<!--
      TorsoPanTransmision:
        motor_offset: -88524
        joint_offset: 0.0
        gear: 158.0
        encoder_res: 20000.0
      TorsoPanTransmisionInv:
        motor_offset: -88524
        joint_offset: 0.0
        gear: 158.0
        encoder_res: 20000.0
        motor_constant: 0.00105

      rightHandMoveAction:
        prefix: right
      leftHandMoveAction:
        prefix: left
      LWRlDiag:
        prefix: left_arm_
      LWRrDiag:
        prefix: right_arm_

      SplineTrajectoryActionJoint:
        joint_names: [torso_0_joint, right_arm_0_joint, right_arm_1_joint, right_arm_2_joint, right_arm_3_joint, right_arm_4_joint, right_arm_5_joint, right_arm_6_joint, left_arm_0_joint, left_arm_1_joint, left_arm_2_joint, left_arm_3_joint, left_arm_4_joint, left_arm_5_joint, left_arm_6_joint]
        lower_limits: [-2.0, -2.96, -2.09, -2.96, -2.09, -2.96, -2.09, -2.96, -2.96, -2.09, -2.96, -2.09, -2.96, -2.09, -2.96]
        upper_limits: [2.0, 2.96, 2.09, 2.96, 2.09, 2.96, 2.09, 2.96, 2.96, 2.09, 2.96, 2.09, 2.96, 2.09, 2.96]
      HeadSplineTrajectoryActionJoint:
        joint_names: [head_pan_joint, head_tilt_joint]
        lower_limits: [-1.57, -1]
        upper_limits: [1.57, 1]

      TfPub:
        frame_ids: [right_arm_7_link, left_arm_7_link, torso_base, torso_base, torso_base, torso_base]
        child_frame_ids: [right_arm_tool, left_arm_tool, right_arm_cmd, left_arm_cmd, right_arm_tool_measured, left_arm_tool_measured]
-->
  </group>

</launch>
